lib/supplejack_api/index_processor.rb

```diff
@@ -2,28 +2,46 @@
 
 module SupplejackApi
   class IndexProcessor
-    attr_accessor :batch_size
+    attr_accessor :size
 
-    def initialize(batch_size)
-      @batch_size = batch_size || 1000
+    def initialize(size)
+      @size = size || 1000
     end
 
     def call
-      p 'Looking for records to index..'
+      loop do
+        p 'Looking for records to index..'
 
-      records_to_index = SupplejackApi::Record.ready_for_processing.where(status: 'active').limit(batch_size)
+        records_to_index = SupplejackApi::Record.ready_for_indexing.where(status: 'active')
 
-      p "There are #{records_to_index.to_a.count} records to be indexed.."
+        records_to_index.in_groups_of(size).each do |records|
+          p "There are #{records_to_index.to_a.count} records to be indexed.."
 
-      BatchIndexRecords.new(records_to_index).call
+          BatchIndexRecords.new(records).call
+        end
 
-      p 'Looking for records to remove..'
+        p 'Looking for records to remove..'
 
-      records_to_remove = SupplejackApi::Record.ready_for_processing.where(:status.ne => 'active').limit(batch_size)
+        records_to_remove = SupplejackApi::Record.ready_for_indexing.where(:status.ne => 'active')
 
-      p "There are #{records_to_remove.to_a.count} records to be removed from the index.."
+        records_to_remove.in_groups_of(size).each do |records|
+          p "There are #{records_to_remove.to_a.count} records to be removed from the index.."
 
-      BatchRemoveRecordsFromIndex.new(records_to_remove).call
+          BatchRemoveRecordsFromIndex.new(records).call
+        end
+
+        p 'Checking if remaining records are above the threshold...'
+
+        records_to_index = SupplejackApi::Record.ready_for_indexing.where(status: 'active')
+        records_to_remove = SupplejackApi::Record.ready_for_indexing.where(:status.ne => 'active')
+
+        if records_to_index.count < 100 && records_to_remove.count < 100
+          p 'There are not enough records at the moment. Will try again soon...'
+          break
+        else
+          p 'Continuing to index records...'
+        end
+      end
```

Comment:
Hmm ok, I see what you did here.. although not I'm not so sure you still need my idea of a loop.
Using no `limit` and using `in_groups_of` is much more elegant. 

I think you should remove the loop and the break, it's no longer needed.

Reply:
Nice, I was wondering about the loop! Updated now